bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,86 @@
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Emojis, Roles, STAFF_ROLES
+from bot.converters import Expiry
+from bot.utils.scheduling import Scheduler
+from bot.utils.time import format_infraction_with_duration
+
+# Constant error messages
+TIME_FORMAT_NOT_VALID = f"{Emojis.cross_mark}Please specify a valid time format ex. 10h or 1day."
+TIME_LESS_EQ_0 = f"{Emojis.cross_mark}Duration can not be a 0 or lower."
+USER_ALREADY_ALLOWED_TO_STREAM = f"{Emojis.cross_mark}This user can already stream."
+USER_ALREADY_NOT_ALLOWED_TO_STREAM = f"{Emojis.cross_mark}This user already can't stream."
+
+
+class Stream(commands.Cog):
+    """Grant and revoke streaming permissions from users."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.scheduler = Scheduler(self.__class__.__name__)
+
+    @staticmethod
+    async def _remove_streaming_permission(schedule_user: discord.Member) -> None:
+        """Remove streaming permission from Member."""
+        await schedule_user.remove_roles(discord.Object(Roles.video), reason="Temporary streaming access revoked")
+
+    @commands.command(aliases=("streaming",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(
```

Comment:
We'll want to add a default duration for the command.  We're thinking 30 minutes would be a good one.  Along with that, we'll need that default added to the config.yaml and the constants file.

Reply:
done in commit [074ce91205539ca06c1c048c62b7e649c4ae78b5](https://github.com/python-discord/bot/pull/1303/commits/074ce91205539ca06c1c048c62b7e649c4ae78b5) and [a0e2179da398a9164582e7debb7e139754434385](https://github.com/python-discord/bot/pull/1303/commits/a0e2179da398a9164582e7debb7e139754434385)